hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/OzoneBucket.java

```diff
@@ -563,7 +583,7 @@ public long getUsedNamespace() {
   public Iterator<? extends OzoneKey> listKeys(String keyPrefix,
       String prevKey) throws IOException {
 
-    if(OzoneFSUtils.isFSOptimizedBucket(getMetadata())){
+    if(OzoneFSUtils.isFSOptimizedBucket(bucketType)){
```

Comment:
should we not check bucketTypre.equals(FSO) or something like that ?

Reply:
can we have a KeyIteratorFactory that returns KeyIteratorWithFSO, KeyIterator or future KeyIterator types properly?

Reply:
@mukul1987, Now we checking using bucketType.equals(BucketType.FILE_SYSTEM_OPTIMIZED).
@xiaoyuyao Implemented KeyIteratorFactory that returns KeyIteratorWithFSO, KeyIterator.